Classes/ElasticSearchClient.php

```diff
@@ -28,15 +28,76 @@
 class ElasticSearchClient extends \Flowpack\ElasticSearch\Domain\Model\Client
 {
     /**
+<<<<<<< HEAD
+     * @var DimensionsService
+     * @Flow\Inject
+     */
+    protected $dimensionsService;
+
+    /**
+     * The index name to be used for querying (by default "neoscr")
+     *
+     * @var string
+     */
+    protected $indexName;
+
+    /**
+     * MD5 hash of the content dimensions
+     *
+     * @var string
+     */
+    protected $dimensionsHash;
+
+    /**
+     * @var array
+     */
+    protected $dimensions = [];
+
+    /**
      * @var IndexNameStrategyInterface
      * @Flow\Inject
      */
     protected $indexNameStrategy;
 
+    /**
+     * @param string $dimensionsHash
+     */
+    public function setDimensions(array $dimensionValues = null)
+    {
+        $dimensionValues = $dimensionValues === null ? [] : $dimensionValues;
+        if ($dimensionValues === []) {
+            $this->dimensions = [];
+            $this->dimensionsHash = null;
+            return;
+        }
+        $this->dimensionsHash = $this->dimensionsService->hash($dimensionValues);
+    }
+
```

Comment:
The called  method getIndexNamePrefix is missing, I guess it got lost with a merge.